src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -144,6 +160,515 @@ TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
+  file_system::CreateFile(full_file_path);
+  const std::vector<uint8_t> data(10, 10);
+  file_system::Write(full_file_path, data);
+  req->Run();
+  file_system::RemoveDirectory(app_storage_folder);
+}
+
+TEST_F(SetAppIconRequestTest,
+       CopyToIconStorage_NonExistentApp_ApplicationNotRegistered) {
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg_vr);
+
+  MockAppPtr mock_app = CreateMockApp();
+  MockAppPtr mock_app_nonexistent;
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app))
+      .WillOnce(Return(mock_app_nonexistent));
+
+  MockApplicationManagerSettings mock_app_mngr_settings;
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillRepeatedly(ReturnRef(mock_app_mngr_settings));
+
+  const std::string app_storage_folder = "./storage";
+  EXPECT_CALL(mock_app_mngr_settings, app_storage_folder())
+      .WillRepeatedly(ReturnRef(app_storage_folder));
+
+  const std::string app_folder = "Test_app";
+  EXPECT_CALL((*mock_app), folder_name()).WillRepeatedly(Return(app_folder));
+
+  MockProtocolHandler mock_protocol_hendler;
+  EXPECT_CALL(app_mngr_, protocol_handler())
+      .WillRepeatedly(ReturnRef(mock_protocol_hendler));
+
+  MockProtocolHandlerSettings mock_protocol_hendler_settings;
+  EXPECT_CALL(mock_protocol_hendler, get_settings())
+      .WillRepeatedly(ReturnRef(mock_protocol_hendler_settings));
+
+  const protocol_handler::MajorProtocolVersion mock_app_mngr_protocol_version =
+      protocol_handler::PROTOCOL_VERSION_4;
+  EXPECT_CALL(mock_protocol_hendler_settings, max_supported_protocol_version())
+      .WillRepeatedly(Return(mock_app_mngr_protocol_version));
+
+  const std::string icon_folder = app_storage_folder + "/" + app_folder;
+  EXPECT_CALL(mock_app_mngr_settings, app_icons_folder())
+      .WillRepeatedly(ReturnRef(icon_folder));
+
+  const uint32_t storage_max_size = 10u;
+  EXPECT_CALL(mock_app_mngr_settings, app_icons_folder_max_size())
+      .WillRepeatedly(ReturnRef(storage_max_size));
+
+  EXPECT_CALL((*mock_app), policy_app_id()).Times(0);
+
+  const std::string directory = app_storage_folder + "/" + app_folder;
+  const std::string full_file_path = directory + "/" + kSyncFileName;
+  file_system::CreateDirectoryRecursively(directory);
+  file_system::CreateFile(full_file_path);
+  req->Run();
+  file_system::RemoveDirectory(app_storage_folder);
+}
+
+TEST_F(SetAppIconRequestTest,
+       CopyToIconStorage_PolicyAppIdEqualFolderName_CantCreateIcon) {
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg_vr);
+
+  MockAppPtr mock_app = CreateMockApp();
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillRepeatedly(Return(mock_app));
+
+  MockApplicationManagerSettings mock_app_mngr_settings;
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillRepeatedly(ReturnRef(mock_app_mngr_settings));
+
+  const std::string app_storage_folder = "./storage";
```

Comment:
@BSolonenko Also repeated almost in every test case. Move it to class or to nameless namespace